src/dev.rs

```diff
@@ -64,12 +64,8 @@ pub enum VerifyFailure {
     },
     /// A permutation did not preserve the original value of a cell.
     Permutation {
```

Comment:
Maybe rename this to `Equality`?

Reply:
@str4d and I discussed that in a pairing, apparently there are many changes we'd like to make to the mock prover and I don't want this PR to get bogged down in those improvements.